packaging/test_rule_engine_plugin_metadata_guard.py

```diff
@@ -287,6 +287,31 @@ def test_plugin_cannot_be_bypassed_with_addw__issue_40(self):
             # Clean up.
             self.rods.assert_icommand(['imeta', 'rm', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
 
+    def test_plugin_cannot_be_bypassed_via_imeta_mod__issue_58(self):
+        config = IrodsConfig()
+
+        # Set JSON configuration for the root collection.
+        root_coll = os.path.join('/', self.admin.zone_name)
+        json_config = json.dumps({
+            'prefixes': ['irods::'],
+            'admin_only': True
+        })
+        self.rods.assert_icommand(['imeta', 'set', '-C', root_coll, self.metadata_guard_attribute_name(), json_config])
+
+        try:
+            with lib.file_backed_up(config.server_config_path):
+                self.enable_rule_engine_plugin(config)
+
+                self.user.assert_icommand(['imeta', 'add', '-C', self.user.session_collection, 'unprotected::issue58', 'v'])
+                self.user.assert_icommand(['imeta', 'mod', '-C', self.user.session_collection, 'unprotected::issue58', 'v', 'n:irods::issue58'], 'STDERR', ['CAT_INSUFFICIENT_PRIVILEGE_LEVEL'])
+
+                # Same as above, but with units
+                self.user.assert_icommand(['imeta', 'add', '-C', self.user.session_collection, 'unprotected::withunits', 'v', 'u'])
+                self.user.assert_icommand(['imeta', 'mod', '-C', self.user.session_collection, 'unprotected::withunits', 'v', 'n:irods::withunits', 'u'], 'STDERR', ['CAT_INSUFFICIENT_PRIVILEGE_LEVEL'])
```

Comment:
Let's duplicate L310 and swap the `n:<new_attr>` and `u`. That'll help increase confidence in the implementation of the metadata API.

Let's make the current assertion the last one in the set of duplicates.

Reply:
Realized the old L310 wasn't doing what I thought it was doing, so I made it do a units update instead (which is what I want, I think?)
Updated now

Reply:
Hmm. Are we not confident in the behavior/test?

Reply:
Nevermind, I mixed up the meaning. Should be resolved now.